### PR TITLE
[ticket/12479] Remove deprecated functions

### PR DESCRIPTION
**Pull request by marc1706**
Related documentation PR:
https://github.com/phpbb/documentation/pull/292

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket:

https://tracker.phpbb.com/browse/PHPBB-12479

### CODE REVIEW

File: phpBB/includes/compatibility_globals.php [hunk 1]
```diff
@@ -49,8 +49,6 @@ function register_compatibility_globals()
 
 	/* @var $request \phpbb\request\request_interface */
 	$request = $phpbb_container->get('request');
-	// Inject request instance, so only this instance is used with request_var
-	request_var('', 0, false, false, $request);
```
[the remark this answers is not in the document]
Comment (marc1706) [reply phpBB/includes/compatibility_globals.php:53]: The sole purpose was to inject the request instance as static variable for request_var. This is no longer needed.

File: tests/version/version_helper_remote_test.php [hunk 2]
```diff
@@ -224,4 +224,65 @@ public function test_get_versions($input, $valid_data, $expected_return = '', $e
...
 		$this->assertEquals($expected_return, $return);
 	}
+
+	public function test_version_phpbb_com()
+	{
+		$file_downloader = new \phpbb\file_downloader();
+
+		$hostname = 'version.phpbb.com';
+
+		if (!checkdnsrr($hostname, 'A'))
```
[the remark this answers is not in the document]
Comment (marc1706) [reply tests/version/version_helper_remote_test.php:234]: The whole point of this test is to test retrieval of a file

[the remark this answers is not in the document]
Comment (Derky) [reply tests/version/version_helper_remote_test.php:234]: Does that mean if there is a problem with version.phpbb.com, this test will fail for all pull requests meaning we can't merge anything?

[the remark this answers is not in the document]
Comment (marc1706) [reply tests/version/version_helper_remote_test.php:234]: It will fail but we can still merge

**Review by Derky — APPROVED** [review]

**Pull request merged**